internal/runtime/client/client.go

```diff
@@ -192,22 +194,58 @@ func (c *client) CallAllExtensions(ctx context.Context, hook runtimecatalog.Hook
 			return errors.Wrapf(err, "ExtensionHandler %s failed", registration.Name)
 		}
 		tmpResponse := responseObject.(runtimehooksv1.ResponseObject)
+
 		err = c.CallExtension(ctx, hook, registration.Name, request, tmpResponse)
 		// If one of the extension handlers fails lets short-circuit here and return early.
 		if err != nil {
 			return errors.Wrapf(err, "ExtensionHandler %s failed", registration.Name)
 		}
 		responses = append(responses, tmpResponse)
 	}
-	if err := aggregateResponses(responses); err != nil {
-		return errors.Wrap(err, "failed to aggregate responses")
-	}
+
+	// Aggregate all responses into a single response.
+	// Note: we only get here if all the extension handlers succeeded.
+	aggregateSuccessfulResponses(response, responses)
+
 	return nil
 }
 
-func aggregateResponses(responses []runtimehooksv1.ResponseObject) error {
-	// TODO:(killianmuldoon) implement proper aggregation logic.
-	return nil
+// aggregateSuccessfulResponses aggregates all successful responses into a single response.
+func aggregateSuccessfulResponses(aggregatedResponse runtimehooksv1.ResponseObject, responses []runtimehooksv1.ResponseObject) {
+	// At this point the Status should always be ResponseStatusSuccess and the Message should be empty.
+	// So let's set those values to avoid keeping values that could have been set by the caller of CallAllExtensions.
+	aggregatedResponse.SetMessage("")
+	aggregatedResponse.SetStatus(runtimehooksv1.ResponseStatusSuccess)
+
+	if _, ok := aggregatedResponse.(runtimehooksv1.RetryResponseObject); !ok {
+		// If the aggregated response is not a RetryResponseObject then we're done.
+		return
+	}
+
+	// Note: as all responses have the same type we can assume now that
+	// they all implement the RetryResponseObject interface.
+	aggregatedRetryableResponse := aggregatedResponse.(runtimehooksv1.RetryResponseObject)
```

Comment:
nit, in order to avoid one cast operation
```suggestion
       aggregatedRetryableResponse, ok := aggregatedResponse.(runtimehooksv1.RetryResponseObject)
	if !ok {
		// If the aggregated response is not a RetryResponseObject then we're done.
		return
	}
```